aws_lambda_powertools/event_handler/api_gateway.py

```diff
@@ -217,6 +217,30 @@ def to_dict(self, origin: Optional[str]) -> Dict[str, str]:
             headers["Access-Control-Allow-Credentials"] = "true"
         return headers
 
+    @staticmethod
+    def extract_origin_header(resolver_headers: Dict):
+        """
+        Extracts the 'origin' or 'Origin' header from the provided resolver headers.
+
+        The 'origin' or 'Origin' header can be either a single header or a multi-header.
+
+        Args:
+            resolver_headers (Dict): A dictionary containing the headers.
+
+        Returns:
+            Union[str, List[str], None]: The value(s) of the 'origin' or 'Origin' header.
+                If the header is a single header, a string is returned.
+                If the header is a multi-header, a list of strings is returned.
+                If the header is not present, None is returned.
+        """
+        resolved_header = resolver_headers.get("origin") or resolver_headers.get("Origin")
```

Comment:
Maybe it should be similar to

https://github.com/aws-powertools/powertools-lambda-python/blob/fc376f90a0779226adac6c8ee6619c30b6b2f11d/aws_lambda_powertools/utilities/data_classes/shared_functions.py#L75-L80

Reply:
I refactored this and now using this function: https://github.com/aws-powertools/powertools-lambda-python/blob/fc376f90a0779226adac6c8ee6619c30b6b2f11d/aws_lambda_powertools/utilities/data_classes/shared_functions.py#L42

Reply:
Probably `get_header_value` needs now some extra overloads?